src/main/java/pixelperfect/Spaceship.java

```diff
@@ -15,6 +16,7 @@
 public class Spaceship {
 
 	private static final Spaceship ship = new Spaceship();
+	private RouteGenerator routeGenerator;
```

Comment:
You have a point about the Route not becoming a data class. I will explain my thoughts on the latter part of your comment later, for now this is fine.